src/queries/decimalsForToken.ts

```diff
@@ -10,20 +13,24 @@ export function weiPerToken(decimals: BigNumberish): BigNumber {
 export const useDecimalCountForToken =
   buildQueryHookWhenParamsDefinedChainAddrs<
     number,
-    [_p1: "erc20", assetAddress: string | NATIVE_TOKEN | undefined, _p2: "decimals"],
-    [assetAddress: string | NATIVE_TOKEN]
+    [
+      _p1: "erc20",
+      asset: ReserveOrNativeTokenDefinition | undefined,
```

Comment:
This will always compare non-equal for native tokens as Symbol types are not supported by the `react-query` cache. I have a hack in place that supports them at the top level, but using them as object members will not produce proper caching behaviour.

See how they were implemented in the few queries that supported them previously- wherein it translated them to just the address instead of the entire object.

Reply:
I have moved this into `queries-reserveornative` branch